iaso/api/mobile/groups.py

```diff
@@ -0,0 +1,66 @@
+from django.db.models.query import QuerySet
+from drf_yasg import openapi
+from drf_yasg.utils import swagger_auto_schema
+from rest_framework import permissions, serializers
+from rest_framework.generics import get_object_or_404
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from iaso.api.common import ModelViewSet
+from iaso.api.query_params import APP_ID
+from iaso.api.serializers import AppIdSerializer
+from iaso.models import Group, Project
+
+
+class MobileGroupSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Group
+        fields = [
+            "id",
+            "name",
+        ]
+
+
+class MobileGroupsViewSet(ModelViewSet):
+    """Groups API for Mobile.
+
+    Allows to retrieve a list of groups from the API.
+
+    Returns a lighter payload adapted for the mobile application.
+
+    `GET /api/mobile/groups/?app_id=some.app.id`
+    """
+
+    http_method_names = ["get", "head", "options"]
+    permission_classes = [permissions.AllowAny]
+    results_key = "groups"
```

Comment:
I have just realized that as I was going to implement the API, the ticket asks for the list to be returned directly but here the list is returned inside an object with a "groups" key

Reply:
OK I'll fix that. It seemed like the standard way of returning lists in Iaso endpoints.

Reply:
The problem here is that it should depend on a feature flag that we have. If the project that you are using requires authentication, this endpoint should require authentication. 

Other than that, I think group list is something that anybody connected should be able to access.

Reply:
So, it does not require a specific authorization, access to this is allowed 
- either if the project does not require authentication
- or if the project requires authentication and you are authentified

(projects are linked to the app_id )